test/statistics_listener.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import sys
+from threading import Lock
+from typing import List
+
+from metrics_statistics_msgs.msg import MetricsMessage
+
+import rclpy
+from rclpy.node import Node
+from rclpy.task import Future
+
+# QoS history_depth to use for listener node
+QOS_DEPTH = 10
```

Comment:
Maybe we should use the full depth as mentioned on rosbag2 and rclcpp PRs in test.

Reply:
I think that should happen together with changing the depth to `KeepAll()` everywhere in the code. I would refrain from changing the value here and having it at 10 everywhere else, in order to avoid potential mismatches.